File: Practicos/1/NewtonRaphson/NewtonRaphson.py
from math import sin, cos

#Define una equacion cualquiera
def f(x):
    return sin(x) #PlaceHolder

#Define la derivada de la equacion anterior
def fd(x) :
    return cos(x)

if __name__ == "__main__":
    Xo = float(input("Ingrese un punto inicial: ")) #Lee el punto inicial
    acc = float(input("Introduzca la presicion de la estimacion: ")) #Lee la presicion

    if abs(fd(Xo)) < 1: #Comprueba la condicion de convergencia del metodo
        print("En el intervalo hay al menos una raiz")

        X1 = f(Xo) #Evaluo la ecuacion y su derivada en el punto inicial
        while abs(X1-Xo) >= acc:
            Xo = X1 #Evaluo la ecuacion y su derivada en el punto inicial
            X1 = Xo-f(Xo)/fd(Xo) #Evaluo la ecuacion y su derivada en el punto inicial

        print("El valor es: " + str(Xo))
    else:
        print("No hay raices")

File: Practicos/1/NewtonRaphson/NewtonRaphson.cpp
#include<iostream>
#include<cmath>
using namespace std;

//Define una equacion cualquiera
float f(float x) {
	return sin(x); //PlaceHolder
}

//Define la derivada de la equacion anterior
float fd(float x) {
	return cos(x);
}

int main() {
	float Xo;
	float acc;
	
	std::cout << "Ingrse un punto inicial ";
	std::cin >> Xo; //Lee el punto inicial
	
	std::cout << "Introduzca la presicion ";
	std::cin >> acc; //Lee la presicion
	
	if(abs(fd(Xo)) < 1) { //Comprueba la condicion de convergencia del metodo
		std::cout << "En el intervalo hay al menos una raiz" << '\n';
		
		float X1 = f(Xo); //Evaluo la ecuacion y su derivada en el punto inicial
		while (abs(X1-Xo) >= acc) {
			Xo = X1; //Evaluo la ecuacion y su derivada en el punto inicial
			X1 = Xo-f(Xo)/fd(Xo); //Evaluo la ecuacion y su derivada en el punto inicial
		}
		
		std::cout << "El valor es: " << Xo << '\n';
	}
	else {
		std::cout << "No hay raices" << '\n';
	}
	
	return 0;
}
